backend/mem/location.go

```diff
@@ -0,0 +1,224 @@
+package mem
+
+import (
+	"errors"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Location implements the vfs.Location interface specific to in-memory FileSystem.
+type Location struct {
+	exists     bool
+	firstTime  bool
+	name       string //the path that this location exists on
+	fileSystem *FileSystem
+	volume     string
+}
+
+//String implements io.Stringer by returning the location's URI as a string
+func (l *Location) String() string {
+
+	return l.URI()
+}
+
+//List finds all of the files living at the current location and returns them in a slice of strings.
+//If there are no files at location, then an empty slice will be returned
+func (l *Location) List() ([]string, error) {
+
+	str := l.Path()                      //full path of this location
```

Comment:
I'm curious why in some functions you comment every line and others there are no comments at all. Over time you'll find a useful balance. As a general rule, simple assignments don't really need comments (if they do, you may want to rethink your variable names, in this case "str", presumably "string", is the name of a data type, not the actual contents, which is a path... "p" for path is more descriptive than a generic data type. (single character names of course should be limited to small functions or code blocks... if this was something used in a larger area, I would recommend "filePath" or similar over just "p".